### PR TITLE
Add config API description

### PR DESCRIPTION
**Pull request by tengqm**
closes: #211

### CODE REVIEW

**Comment by tengqm** [comment]
/hold
depends on: #176
/approve cancel

File: genref/config.yaml [hunk 1]
```diff
@@ -56,90 +61,108 @@ apis:
...
   # pkg only contains encryption config. Need to differentiate it from the
   # apiserver-config above.
   - name: apiserver-encryption
+    title: kube-apiserver encryption configuration (v1)
     package: k8s.io/apiserver
     path: pkg/apis/config/v1
 
     # Note: This is named apiserver-audit because it is part of the apiserver module
   - name: apiserver-audit
+    title: kube-apiserver audit configuration (v1)
```
Comment (kbhawkey) [inline genref/config.yaml:115]: Hi @tengqm .
Is this new field the page (H1 level heading) title?
I assume readers can match the package and path to the title.
The generated page (https://deploy-preview-26606--kubernetes-io-master-staging.netlify.app/docs/reference/config-api/apiserver-audit.v1/#audit-k8s-io-v1-Level) uses the "DisplayName"?

File: genref/config.yaml [hunk 2]
```diff
@@ -56,90 +61,108 @@ apis:
...
     package: k8s.io/kube-proxy
     path: config/v1alpha1
 
   - name: metrics
+    title: ""
     package: k8s.io/metrics
     path: pkg/apis/metrics/v1alpha1
 
   - name: metrics
+    title: ""
```
Comment (kbhawkey) [inline genref/config.yaml:74]: Why is this title empty?

File: genref/config.yaml [hunk 3]
```diff
@@ -56,90 +61,108 @@ apis:
...
     package: k8s.io/metrics
     path: pkg/apis/metrics/v1beta1
 
   - name: client-authentication
+    title: Client authentication (v1alpha1)
```
Comment (kbhawkey) [inline genref/config.yaml:79]: nit: Does it make sense to capitalize the titles (the site typically does capitalize titles, but this case seems different) or leave lowercase?

**Comment by kbhawkey** [comment]
Have a couple of questions, otherwise LGTM.
/approve

File: genref/config.yaml (hunk 1 above)
Comment (tengqm) [reply genref/config.yaml:115]: Yes, the `title` field will be served as the "title".
The generated pages are not yet updated.

File: genref/config.yaml (hunk 2 above)
Comment (tengqm) [reply genref/config.yaml:74]: Not sure what to use for this now. Can propose a temp title.

File: genref/config.yaml (hunk 3 above)
Comment (tengqm) [reply genref/config.yaml:79]: OK.

**Comment by kbhawkey** [comment]
This seems ready to go. The title strings can be updated as needed.
Thanks @tengqm !
/lgtm

**Pull request merged**